File: example/ShumachTurnipSample/ShumachTurnipSample/PatternB_1ViewController.m
//
//  PatternB_1ViewController.m
//  ShumachTurnipSample
//
//  Created by shusaku FUKUMINE on 2014/03/07.
//  Copyright (c) 2014年 shusaku FUKUMINE. All rights reserved.
//

#import "PatternB_1ViewController.h"


@interface PatternB_1ViewController (PrivateMethods)
- (void)pushedNextButton:(id)sender;
- (void)patternB_2ViewControllerDidFinish;
@end


@implementation PatternB_1ViewController


- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
  self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
  if (self) {
    // Custom initialization
  }

  return self;
}


- (void)dealloc
{
  if (_patternB_2ViewController != nil) {
    [self.navigationController popViewControllerAnimated:NO];
    [_patternB_2ViewController release];
    _patternB_2ViewController = nil;
  }

  [super dealloc];
}


- (void)viewDidLoad
{
  [super viewDidLoad];
  // Do any additional setup after loading the view.

  _nextButton = [[UIBarButtonItem alloc]
    initWithTitle:@"Next"
    style:UIBarButtonItemStyleBordered
    target:self
    action:@selector(pushedNextButton:)];

  self.navigationItem.rightBarButtonItem = _nextButton;
}


- (void)didReceiveMemoryWarning
{
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}


//////////////////////////////////////////////////////////////////////////////////////////////////
// PrivateMethods

- (void)pushedNextButton:(id)sender
{
  if (_patternB_2ViewController != nil) {
    [_patternB_2ViewController release];
    _patternB_2ViewController = nil;
  }

  _patternB_2ViewController = [[PatternB_2ViewController alloc] initWithNibName:nil bundle:nil];
  _patternB_2ViewController.delegate = (id<PatternB_2ViewControllerDelegate>)self;
  [self.navigationController pushViewController:_patternB_2ViewController animated:NO];
}


- (void)patternB_2ViewControllerDidFinish
{
  [self.navigationController popViewControllerAnimated:NO];

  [_patternB_2ViewController release];
  _patternB_2ViewController = nil;
}


@end

File: example/ShumachTurnipSampleSwift/ShumachTurnipSampleSwift/ShumachTurnipSampleSwift/PatternB_1ViewController.swift
//
//  PatternB_1ViewController.swift
//  ShumachTurnipSampleSwift
//
//  Created by shusaku FUKUMINE on 2014/06/12.
//  Copyright (c) 2014年 shusaku FUKUMINE. All rights reserved.
//

import UIKit


class PatternB_1ViewController: UIViewController, PatternB_2ViewControllerDelegate {


  var patternB_2ViewController: PatternB_2ViewController?


  init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
    super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    // Custom initialization

    self.view.backgroundColor = UIColor.whiteColor()
  }


  override func viewDidLoad() {
    super.viewDidLoad()

    // Do any additional setup after loading the view.

    var nextButton = UIBarButtonItem(title: "Next", style: .Bordered, target: self, action: "pushedNextButton:")

    self.navigationItem.rightBarButtonItem = nextButton;
  }


  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }
    

  func reset() {

    self.navigationController.popViewControllerAnimated(false)
    self.patternB_2ViewController = nil
  }


  func pushedNextButton(barButtonItem: UIBarButtonItem) {

    self.patternB_2ViewController = PatternB_2ViewController(nibName: nil, bundle: nil)
    self.patternB_2ViewController!.delegate = self
    self.navigationController.pushViewController(patternB_2ViewController, animated: false)

  }


  func patternB_2ViewControllerDidFinish() {
    
    self.navigationController.popViewControllerAnimated(false)
    self.patternB_2ViewController = nil

  }

}
